proxy/src/session.rs

```diff
@@ -47,14 +61,24 @@ pub async fn current<R: registry::Client>(
 ) -> Result<Session, error::Error> {
     let mut session = get(store, KEY_CURRENT)?;
 
+    // Reset the permissions
+    session.permissions.register_handle = false;
```

Comment:
```suggestion
    session.permissions.register_handle = true;
```

If set to true and in combination with the other suggestion you can drop the last else branch.

Reply:
I find it easier to read if all permissions are reset to the same value and that permissions are explicitly set to `true`. If we set `register_handle` to true here, the last else branch could be dropped as you suggest, but for consistency we would have to add on for the outermost if checking if an identity exists. If no identity is set in the session, registering a handle should not be permitted.

Reply:
That's a good point, it would move the complexity to another place, I think it's not worth changing then. What you could try tho is to use the default of the struct, see comment below.